xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java

```diff
@@ -87,17 +87,20 @@ public void onComment(String content, TagStack stack)
         // If the comment starts with "startwikilink" then we need to gather all XHTML tags inside
         // the A tag, till we get a "stopwikilink" comment.
         // Same for "startimage" and "stopimage".
-        if (!ignoreElements && content.startsWith("startwikilink:")) {
-            handleLinkCommentStart(XMLUtils.unescapeXMLComment(content), stack);
-        } else if (!ignoreElements && content.startsWith("stopwikilink")) {
-            handleLinkCommentStop(stack);
-        } else if (!ignoreElements && content.startsWith("startimage:")) {
-            handleImageCommentStart(XMLUtils.unescapeXMLComment(content), stack);
-        } else if (!ignoreElements && content.startsWith("stopimage")) {
-            handleImageCommentStop(stack);
-        } else if (!ignoreElements && content.startsWith("startmacro")) {
-            super.onComment(XMLUtils.unescapeXMLComment(content), stack);
-        } else {
+        if(!ignoreElements) {
+            if (content.startsWith("startwikilink:")) {
+                handleLinkCommentStart(XMLUtils.unescapeXMLComment(content), stack);
+            } else if (content.startsWith("stopwikilink")) {
+                handleLinkCommentStop(stack);
+            } else if (content.startsWith("startimage:")) {
+                handleImageCommentStart(XMLUtils.unescapeXMLComment(content), stack);
+            } else if (content.startsWith("stopimage")) {
+                handleImageCommentStop(stack);
+            } else (content.startsWith("startmacro")) {
+                super.onComment(XMLUtils.unescapeXMLComment(content), stack);
+            }
+        }
+        else {
             super.onComment(content, stack);
```

Comment:
The way you changed the id this is not going to be executed in all places it's supposed to.

Reply:
which id?

Reply:
I just made a parent if condition, which it was going to check anyway multiple times.